examples/medplum-chart-demo/src/bots/core/charting-utils.ts

```diff
@@ -0,0 +1,281 @@
+import { getReferenceString } from '@medplum/core';
+import {
+  Bundle,
+  BundleEntry,
+  ClinicalImpression,
+  Coding,
+  Condition,
+  Encounter,
+  Observation,
+  ObservationComponent,
+  Patient,
+  Practitioner,
+  Quantity,
+  Reference,
+  Resource,
+} from '@medplum/fhirtypes';
+import { randomUUID } from 'crypto';
+
+export interface ObservationData {
+  bloodPressure: {
+    systolic?: number;
+    diastolic?: number;
+  };
+  height?: Quantity;
+  weight?: Quantity;
+  bmi?: Quantity;
+  date?: string;
+}
+
+export interface ConditionData {
+  reasonForVisit: Coding;
+  problemList: boolean;
+}
+
+export interface ClinicalImpressionData {
+  visitLength?: number;
+  assessment?: string;
```

Comment:
In general, we want to reduce the number of custom types, as it adds cognitive overhead for new readers to the code. 

I've made a number of comments below about eliminating these and instead using them as discrete input parameters to the functions below